rholang/src/test/scala/coop/rchain/rholang/interpreter/ReduceSpec.scala

```diff
@@ -2059,4 +2056,284 @@ class ReduceSpec extends FlatSpec with Matchers with PersistentStoreTester {
     result.exprs should be(Seq(Expr(resultList)))
     errorLog.readAndClearErrorVector().unsafeRunSync should be(Vector.empty[InterpreterError])
   }
+  "toSet" should "transform [1, 2, 3] into Set(1, 2, 3)" in {
+
+    implicit val errorLog: ErrorLog[Task] = new ErrorLog[Task]()
+    val toSetCall: EMethod =
+      EMethod(
+        "toSet",
+        EListBody(
+          EList(
+            List[Par](
+              GInt(1L),
+              GInt(2L),
+              GInt(3L)
+            )
+          )
+        ),
+        List[Par]()
+      )
+
+    val result: Par = withTestSpace(errorLog) {
+      case TestFixture(_, reducer) =>
+        implicit val env: Env[Par] = Env[Par]()
+        val toSetTask             = reducer.evalExpr(toSetCall)
+        Await.result(toSetTask.runToFuture, 3.seconds)
+    }
+    val resultList =
+      ESetBody(
+        ParSet(
+          List(
+            GInt(1L),
+            GInt(2L),
+            GInt(3L)
+          )
+        )
+      )
+    result.exprs should be(Seq(Expr(resultList)))
+    errorLog.readAndClearErrorVector().unsafeRunSync should be(Vector.empty[InterpreterError])
+  }
+
+
+  it should "transform [] into Set()" in {
+    implicit val errorLog: ErrorLog[Task] = new ErrorLog[Task]()
+    val toSetCall: EMethod =
+      EMethod(
+        "toSet",
+        EListBody(
+          EList(
+            List[Par](
+            )
+          )
+        ),
+        List()
```

Comment:
All those tests have the same structure. It feels as if we could reduce them all to:

`checkResult("@0!([1,2,3].toSet())", "@0!(Set(1, 2, 3))")`

Or, equally well, we could implement the test in RhoSpec.

Either way, there's duplication between the tests to be removed.

Reply:
Ok, there are error cases - that's another test util method

Reply:
indeed, there is some refactoring required for the ReduceSpec tests. I'll try to address the duplication.